tests/string_test.py

```diff
@@ -441,7 +447,24 @@ def test_peel(self):
         with self.assertRaises(ValueError):
             run_test_peel(self.gremlins_strings, self.gremlins_test_strings, '')  
         run_test_peel(self.gremlins_strings, self.gremlins_test_strings, '"')  
-        run_test_peel(self.gremlins_strings, self.gremlins_test_strings, ' ') 
+        run_test_peel(self.gremlins_strings, self.gremlins_test_strings, ' ')
+
+        # Run a test with a specific set of strings to verify strings.bytes matches expected output
+        series = pd.Series(["k1:v1", "k2:v2", "k3:v3", "no_colon"])
+        pda = ak.from_series(series, "string")
+
+        # Convert Pandas series of strings into a byte array where each string is terminated by a null byte.
+        # This mimics what should be stored server-side in the strings.bytes pdarray
+        expected_series_dec = convert_to_ord(series.to_list())
+        actual_dec = pda.bytes.to_ndarray().tolist()
+        self.assertEqual(expected_series_dec, actual_dec)
```

Comment:
I'm not familiar enough with pytest to know this, but does `self.assertEqual(list1, list2)` actually do the element-wise comparison and check that they are all `True`? To me, I would be wary that it is just calling `assert list1 == list2` under the hood, which would produce the wrong result (or an error).

Elsewhere, I have been using `self.assertTrue((arr1 == arr2).all())` with either numpy arrays or arkouda arrays to ensure all the element-wise comparisons are happening, but if `assertEqual` does the right thing with python lists, I would be interested to know.

Reply:
That's a good question, I'm fairly certain python's unittest.TestCase (where the assertion methods are coming from) really default down to the operator methods, so assertEquals(a, b) delegates to a == b which is an equivalence test vs `a is b` which is same ref.

I ran the following in a python console to double check
[Gist here](https://gist.github.com/glitch/20db958b8ae5d357ed90cc0586fb09a6)
```
x = ["a", "b", "c"]
y = ["a", "b", "c"]
z = ["c", "b", "a"]
x == y
> True
x is y
> False
x == z
> False

import unittest
t = unittest.TestCase()
t.assertEqual(x, y)  # passes
t.assertTrue(x is y) # fails
t.assertEqual(x, z)  # fails

# There is also a assertListEqual(a, b) which may be a bit more explicit
t.assertListEqual(x, y)  # passes
t.assertListEqual(x, z)  # fails
```

So while it's testing the correct thing, I think I'll push an update to improve clarity with `assertListEquals`

Reply:
Note: I don't have a link to the python src handy, but for 3.7.9 unittest / case.py _baseAssertEqual the code is this
```python
    def _baseAssertEqual(self, first, second, msg=None):
        """The default assertEqual implementation, not type specific."""
        if not first == second:
            standardMsg = '%s != %s' % _common_shorten_repr(first, second)
            msg = self._formatMessage(msg, standardMsg)
            raise self.failureException(msg)
```
so it is basically python's `==` function, we should be good.

Reply:
Ah, ok. I guess I had confused the behavior of python lists with numpy arrays and had forgotten that `a == b` will work with the former. 

Side note: if `a` and `b` are numpy arrays (or arkouda arrays), `a == b` will return a bool array the same length as `a` and `b` with the result of each element-wise comparison, instead of a single bool value. So calling `assertEqual(a, b)` will raise a ValueError error saying "the truth value of an array is ambiguous" or something like that. Not a problem for python lists, but something to watch out for when writing unit tests on numpy/arkouda arrays.